notebooks/1.2-pyc-eda-mix.py

```diff
@@ -0,0 +1,500 @@
+
+train[mix_columns].head()
+
+
+train[mix_columns].isna().mean()
+
+
+train[mix_columns].describe()
+
+
+# # Sales variables
+# 
+# - MoSold
+# - YrSold
+# - SaleType
+# - SaleCondition
+
+# ## Sale Condition
+# 
+# 
+# Condition of sale
+# 
+#        Normal	Normal Sale
+#        Abnorml	Abnormal Sale -  trade, foreclosure, short sale
+#        AdjLand	Adjoining Land Purchase
+#        Alloca	Allocation - two linked properties with separate deeds, typically condo with a garage unit	
+#        Family	Sale between family members
+#        Partial	Home was not completed when last assessed (associated with New Homes)
+# 
+
+multiple_ecdf('SaleCondition')
+
+
+# **Insights**
+# 
+# - The partial category in this column appears to increase significantly the sale Price
+# - Maybe the other categories can be joined
+
+# ## SaleType
+# 
+# 
+# SaleType: Type of sale
+# 		
+#        WD 	Warranty Deed - Conventional
+#        CWD	Warranty Deed - Cash
+#        VWD	Warranty Deed - VA Loan
+#        New	Home just constructed and sold
+#        COD	Court Officer Deed/Estate
+#        Con	Contract 15% Down payment regular terms
+#        ConLw	Contract Low Down payment and low interest
+#        ConLI	Contract Low Interest
+#        ConLD	Contract Low Down
+#        Oth	Other
+
+multiple_ecdf('SaleType')
+
+
+# **Insights**
+# 
+# - have a lot of categories, but appears that just the category New increase in the sale
+# - Maybe we should try an unsupervised model to join these categories
+
+# ## Sale date
+# 
+# Month and Year of the sale
+
+sns.pairplot(train[['SalePrice','MoSold','YrSold']])
+
+
+train[train['YrSold']==2010]['MoSold'].value_counts()
+
+
+fig = make_subplots(rows=1, cols=2,
+                   subplot_titles=("Mean", "Std"))
+
+fig.add_trace(go.Scatter(x=np.sort(train['YrSold'].unique()), 
+                         y=train.groupby(['YrSold'])['SalePrice'].mean()),
+            row=1,col=1)
+
+fig.add_trace(go.Scatter(x=np.sort(train['YrSold'].unique()), 
+                         y=train.groupby(['YrSold'])['SalePrice'].std()),
+            row=1,col=2)
+
+fig.update_layout(showlegend=False, title_text="Mean and Std of Sales Price per Year")
+fig.show()
+
+
+train['day'] = 1
+train['dateSold'] = pd.to_datetime(train.rename(columns={"YrSold": "year",
+                                                         "MoSold": "month"})[['year','month','day']],
+                                   unit='D')
+train.drop(columns=['day'], inplace=True)
+fig = px.line(train.groupby(['dateSold'])[['SalePrice', 'dateSold']].mean(), 
+              x=train.groupby(['dateSold'])[['SalePrice', 'dateSold']].mean().index,
+              y='SalePrice')
+fig.show()
+
+
+# **Insights**
+# - The last month we have just 6 data, so, probably, we don't have data of the full month
+# - Maybe it's better to join the 2 columns to understand the timeline
+# - The timeline appears to make more sense that the Month or Year column alone
```

Comment:
should we drop the year and the month columns ?

Reply:
I don't think so. But it's better to analyze the timeline